src/openrct2/ride/Ride.cpp

```diff
@@ -6401,7 +6401,7 @@ void Ride::UpdateMaxVehicles()
         {
             case RideMode::ContinuousCircuitBlockSectioned:
             case RideMode::PoweredLaunchBlockSectioned:
-                maxNumTrains = std::clamp(num_stations + num_block_brakes - 1, 1, 31);
+                maxNumTrains = std::clamp(num_stations + num_block_brakes - 1, 1, int32_t(MAX_VEHICLES_PER_RIDE));
```

Comment:
if you moved this to the clamp i think it should compile without warning.
`std::clamp<int32_t>`